### PR TITLE
hwmv2: Port NXP iMX RT500 to HWMv2

### PR DESCRIPTION
**Pull request by danieldegrasse**
This PR ports the RT500 SOC and RT595 EVK to HWMv2. This is a dual architecture SOC (Xtensa Fusion F1 and Arm Cortex M33). Therefore, this PR has been separated from the remainder of the iMX RT series ports, since it may require more detailed review.

~*Note*- this PR Depends on #69264, and contains commits from it until it is merged~

### CODE REVIEW

File: soc/nxp/imxrt/rt5xx/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright 2024 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if (CONFIG_SOC_MIMXRT595S_CM33)
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt5xx/CMakeLists.txt:7]:
```suggestion
if(CONFIG_SOC_MIMXRT595S_CM33)
```

File: soc/nxp/imxrt/rt5xx/Kconfig [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	select HAS_MCUX_USB_LPCIP3511
+	select HAS_MCUX_CTIMER
+
+config SOC_MIMXRT595S_F1
+	select XTENSA
+	select XTENSA_HAL if ("$(ZEPHYR_TOOLCHAIN_VARIANT)" != "xcc" && "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "xt-clang")
+	select XTENSA_RESET_VECTOR
+	select XTENSA_USE_CORE_CRT1
+	help
+	  NXP RT5xx ADSP Series
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt5xx/Kconfig:41]: help text for these should be in `Kconfig.soc`

File: soc/nxp/imxrt/rt5xx/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	default n
+
+config CACHE_MANAGEMENT
+	default n
+
+config LOG
+	default n
+
+config SOC_TOOLCHAIN_NAME
+	default "nxp_rt500_adsp"
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt5xx/Kconfig.defconfig:120]: `Kconfig.soc`

File: soc/nxp/imxrt/rt5xx/Kconfig.soc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+config SOC_MIMXRT595S_F1
+	bool
+	select SOC_MIMXRT595S
+
+config SOC_SERIES
+	default "rt5xx" if SOC_SERIES_IMXRT_5XX
+
+config SOC
+	default "mimxrt595s_cm33" if SOC_MIMXRT595S_CM33
+	default "mimxrt595s_dsp" if SOC_MIMXRT595S_F1
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt5xx/Kconfig.soc:27]: default "mimxrt595s" if SOC_MIMXRT595S

File: boards/nxp/mimxrt595_evk/Kconfig [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
 # Copyright (c) 2023 Google LLC.
 # SPDX-License-Identifier: Apache-2.0
 
+config BOARD_INIT_PRIORITY
+	int "Board initialization priority"
+	default 45
+	help
+	  Board initialization priority.
+
+
```
Comment (nordicjm) [inline boards/nxp/mimxrt595_evk/Kconfig:11]:
```suggestion
```

File: boards/nxp/mimxrt595_evk/doc/index.rst [hunk 6]
```diff
@@ -118,7 +118,7 @@ already supported, which can also be re-used on this mimxrt595_evk board:
 
 The default configuration can be found in the defconfig file:
 
-	``boards/arm/mimxrt595_evk/mimxrt595_evk_cm33_defconfig``
+	``boards/arm/mimxrt595_evk/mimxrt595_evk_rt595_cm33_defconfig``
```
Comment (nordicjm) [inline boards/nxp/mimxrt595_evk/doc/index.rst:121]: Link is wrong and should use zephyr_file e.g.
```suggestion
:zephyr_file:`boards/nxp/mimxrt595_evk/mimxrt595_evk_rt595_cm33_defconfig`
```

File: boards/nxp/mimxrt595_evk/doc/index.rst [hunk 7]
```diff
@@ -185,7 +185,7 @@ To build the hello_world sample for the RT500 DSP core:
 
 .. code-block:: shell
 
-   $ west build -b nxp_adsp_rt595 samples/hello_world
+   $ west build -b mimxrt595_evk//f1 samples/hello_world
```
Comment (nordicjm) [inline boards/nxp/mimxrt595_evk/doc/index.rst:188]: Use full board names and identifiers in all examples `mimxrt595_evk/rt595/f1`

File: boards/nxp/mimxrt595_evk/doc/index.rst [hunk 8]
```diff
@@ -249,7 +249,7 @@ Here is an example for the :ref:`hello_world` application. This example uses the
 
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
-   :board: mimxrt595_evk_cm33
+   :board: mimxrt595_evk//cm33
```
Comment (nordicjm) [inline boards/nxp/mimxrt595_evk/doc/index.rst:252]: etc.

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: soc/nxp/imxrt/rt5xx/Kconfig.defconfig (hunk 3 above)
Comment (danieldegrasse) [reply soc/nxp/imxrt/rt5xx/Kconfig.defconfig]: Moved. I am slightly unclear, what is the guidance for using `Kconfig.defconfig` versus `Kconfig.soc` in HWMv2? Is the idea that any Kconfig that is prefixed with "SOC_" should be in `Kconfig.soc`, or are there more general guidelines to follow?

File: soc/nxp/imxrt/rt5xx/Kconfig.soc (hunk 4 above)
Comment (danieldegrasse) [reply soc/nxp/imxrt/rt5xx/Kconfig.soc]: Fixed

File: boards/nxp/mimxrt595_evk/doc/index.rst (hunk 7 above)
Comment (danieldegrasse) [reply boards/nxp/mimxrt595_evk/doc/index.rst]: Changed- I understand being verbose here, but where will we document the "short name" feature of using `mimxrt595_evk//f1`? I find this pretty useful, and want to make sure customers are aware of it.

File: soc/nxp/imxrt/rt5xx/Kconfig.defconfig (hunk 3 above)
Comment (nordicjm) [reply soc/nxp/imxrt/rt5xx/Kconfig.defconfig]: `Kconfig.defconfig` is for overriding defaults in the zephyr tree, `Kconfig.soc` is just for SoC only Kconfigs that are needed by the build system (that means it only has the SoC Kconfig tree loaded, it does not have the rest of the zephyr Kconfig tree loaded)

File: boards/nxp/mimxrt595_evk/doc/index.rst (hunk 7 above)
Comment (nordicjm) [reply boards/nxp/mimxrt595_evk/doc/index.rst]: It will be in the hwmv2 documentation

File: soc/nxp/imxrt/Kconfig.soc [hunk 9]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_IMXRT
+	bool
+
+config SOC_FAMILY
+	default "imxrt" if SOC_FAMILY_IMXRT
```
Comment (nordicjm) [inline soc/nxp/imxrt/Kconfig.soc:8]: Same issue as other pr:
```suggestion
config SOC_FAMILY_NXP_IMXRT
	bool

config SOC_FAMILY
	default "nxp_imxrt" if SOC_FAMILY_NXP_IMXRT
```

File: soc/nxp/imxrt/rt10xx/Kconfig.soc [hunk 10]
```diff
@@ -0,0 +1,238 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_IMXRT_10XX
+	bool
+	select SOC_FAMILY_IMXRT
+
+config SOC_SERIES
+	default "rt10xx" if SOC_SERIES_IMXRT_10XX
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt10xx/Kconfig.soc:9]: name should match:
```suggestion
	default "imxrt_10xx" if SOC_SERIES_IMXRT_10XX
```

File: soc/nxp/imxrt/rt5xx/Kconfig.defconfig [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+# is disabled, hence MEMC is selected when PM_DEVICE
+# is turned on.
+config MEMC
+	default y
+	select MEMC_MCUX_FLEXSPI
+
+endif
+
+config MCUX_CORE_SUFFIX
+	default "_cm33"
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt5xx/Kconfig.defconfig:72]: raised query about this in main pr

File: soc/nxp/imxrt/rt5xx/Kconfig.soc [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+	  NXP RT5xx CM33 core
+
+config SOC_MIMXRT595S_F1
+	bool
+	select SOC_MIMXRT595S
+	help
+	  NXP RT5xx ADSP Series
+
+config SOC_SERIES
+	default "rt5xx" if SOC_SERIES_IMXRT_5XX
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt5xx/Kconfig.soc:27]:
```suggestion
	default "imxrt_5xx" if SOC_SERIES_IMXRT_5XX
```

File: soc/nxp/imxrt/rt5xx/Kconfig.soc [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+config SOC
+	default "mimxrt595s" if SOC_MIMXRT595S
+
+config SOC_TOOLCHAIN_NAME
+	default "nxp_rt500_adsp"
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt5xx/Kconfig.soc:33]: default... if?

File: soc/nxp/imxrt/rt5xx/Kconfig.soc [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+	bool
+
+config SOC_PART_NUMBER_MIMXRT555SFAWC
+	bool
+
+config SOC_PART_NUMBER_MIMXRT595SFAWC
+	bool
+
+config SOC_PART_NUMBER
+	string
```
Comment (nordicjm) [inline soc/nxp/imxrt/rt5xx/Kconfig.soc:54]:
```suggestion
```

File: soc/nxp/imxrt/rt5xx/Kconfig.soc (hunk 12 above)
Comment (danieldegrasse) [reply soc/nxp/imxrt/rt5xx/Kconfig.soc:27]: Changed `CONFIG_SOC_SERIES` to `imxrt5xx` to align with rest of iMX RT series, and updated Kconfig name to `SOC_SERIES_IMXRT5XX`

**Review by nordicjm — APPROVED** [review]

**Comment by nordicjm** [comment]
Tests using old names need to be updated to use the new name

File: soc/nxp/imxrt/imxrt5xx/Kconfig.soc [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+	  NXP RT5xx ADSP Series
+
+config SOC_SERIES
+	default "imxrt5xx" if SOC_SERIES_IMXRT5XX
+
+config SOC
+	default "mimxrt595s" if SOC_MIMXRT595S
+
+config SOC_TOOLCHAIN_NAME
+	default "nxp_rt500_adsp" if SOC_MIMXRT595S_F1
```
Comment (nordicjm) [inline soc/nxp/imxrt/imxrt5xx/Kconfig.soc:33]: needs `string` here

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: soc/nxp/imxrt/imxrt5xx/Kconfig.soc (hunk 15 above)
Comment (danieldegrasse) [reply soc/nxp/imxrt/imxrt5xx/Kconfig.soc:34]: Added

**Comment by danieldegrasse** [comment]
> Tests using old names need to be updated to use the new name

I believe all are, is there something I'm missing?

```
$ grep -rl "mimxrt595_evk_cm33" . --exclude-dir=".git"
./doc/releases/release-notes-3.3.rst
```

**Comment by nordicjm** [comment]
Compliance issue needs addressing. There was a test failure in the previous run

**Comment by danieldegrasse** [comment]
> Compliance issue needs addressing. There was a test failure in the previous run

Fixed now

**Review by nordicjm — APPROVED** [review]

**Pull request merged**